service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -230,20 +230,10 @@ public void run() {
                         MetadbRequest existingRequest =
                                 requestService.getMetadbRequestById(requestMetadata.getIgoRequestId());
                         if (existingRequest == null) {
-                            LOG.info("Request does not already exist - saving to database: "
-                                    + requestMetadata.getIgoRequestId());
-                            // persist and handle new request
-                            MetadbRequest request =
-                                    RequestDataFactory.buildNewRequestFromMetadata(requestMetadata);
-
-                            LOG.info("Publishing new request metadata to " + CMO_REQUEST_UPDATE_TOPIC);
-                            requestService.saveRequest(request);
-                            messagingGateway.publish(request.getIgoRequestId(),
-                                    CMO_REQUEST_UPDATE_TOPIC,
-                                    mapper.writeValueAsString(
-                                          request.getRequestMetadataList()));
+                            // TODO: Log this request
                         } else if (requestService.requestHasMetadataUpdates(
-                                existingRequest.getLatestRequestMetadata(), requestMetadata)) {
+                                                mapper.writeValueAsString(existingRequest),
```

Comment:
ExistingRequestJson follows the universal schema like the requestMetadataUpdate we received

Reply:
[requested change] could we not use `existingRequest.getLatestRequestMetadata().getRequestMetadataJson()`?

Reply:
No because the stored `requestJson` is in the igo metadata schema not the universal schema

Reply:
As long as the two instances of RequestMetadata are being compared against each other then it shouldn't be problematic to determine if there are updates with the existing `requestService.requestHasMetadataUpdates(...)` method.

I still want to test this further but so far I was able to verify an `existingRequest.getLatestRequestMetadata()` against a "published message" containing only an `RequestMetadata` JSON with "updates" and the comparator was still able to handle this case as expected.